tests/test_sagemaker/test_sagemaker_notebooks.py

```diff
@@ -0,0 +1,228 @@
+        "InstanceType": instance_type,
+        "RoleArn": FAKE_ROLE_ARN,
+    }
+    with assert_raises(ClientError) as ex:
+        resp = sagemaker.create_notebook_instance(**args)
+    assert_equal(ex.exception.response["Error"]["Code"], "ValidationException")
+    expected_message = "Value '{}' at 'instanceType' failed to satisfy constraint: Member must satisfy enum value set: [".format(
+        instance_type
+    )
+
+    assert_true(expected_message in ex.exception.response["Error"]["Message"])
+
+
+@mock_sagemaker
+def test_notebook_instance_lifecycle():
+    sagemaker = boto3.client("sagemaker", region_name=TEST_REGION_NAME)
+
+    NAME_PARAM = "MyNotebookInstance"
+    INSTANCE_TYPE_PARAM = "ml.t2.medium"
+
+    args = {
+        "NotebookInstanceName": NAME_PARAM,
+        "InstanceType": INSTANCE_TYPE_PARAM,
+        "RoleArn": FAKE_ROLE_ARN,
+    }
+    resp = sagemaker.create_notebook_instance(**args)
+    assert_true(resp["NotebookInstanceArn"].startswith("arn:aws:sagemaker"))
+    assert_true(resp["NotebookInstanceArn"].endswith(args["NotebookInstanceName"]))
+
+    resp = sagemaker.describe_notebook_instance(NotebookInstanceName=NAME_PARAM)
+    notebook_instance_arn = resp["NotebookInstanceArn"]
+
+    with assert_raises(ClientError) as ex:
+        sagemaker.delete_notebook_instance(NotebookInstanceName=NAME_PARAM)
+    assert_equal(ex.exception.response["Error"]["Code"], "ValidationException")
+    expected_message = "Status (InService) not in ([Stopped, Failed]). Unable to transition to (Deleting) for Notebook Instance ({})".format(
+        notebook_instance_arn
+    )
+    assert_true(expected_message in ex.exception.response["Error"]["Message"])
+
+    sagemaker.stop_notebook_instance(NotebookInstanceName=NAME_PARAM)
+
+    resp = sagemaker.describe_notebook_instance(NotebookInstanceName=NAME_PARAM)
+    assert_equal(resp["NotebookInstanceStatus"], "Stopped")
+
+    sagemaker.start_notebook_instance(NotebookInstanceName=NAME_PARAM)
+
+    resp = sagemaker.describe_notebook_instance(NotebookInstanceName=NAME_PARAM)
+    assert_equal(resp["NotebookInstanceStatus"], "InService")
+
+    sagemaker.stop_notebook_instance(NotebookInstanceName=NAME_PARAM)
+
+    resp = sagemaker.describe_notebook_instance(NotebookInstanceName=NAME_PARAM)
+    assert_equal(resp["NotebookInstanceStatus"], "Stopped")
+
+    sagemaker.delete_notebook_instance(NotebookInstanceName=NAME_PARAM)
+
+    with assert_raises(ClientError) as ex:
+        sagemaker.describe_notebook_instance(NotebookInstanceName=NAME_PARAM)
+    assert_equal(ex.exception.response["Error"]["Message"], "RecordNotFound")
+
+
+@mock_sagemaker
+def test_describe_nonexistent_model():
+    sagemaker = boto3.client("sagemaker", region_name=TEST_REGION_NAME)
+
+    with assert_raises(ClientError) as e:
+        resp = sagemaker.describe_model(ModelName="Nonexistent")
+    assert_true(
+        e.exception.response["Error"]["Message"].startswith("Could not find model")
+    )
+    pass
```

Comment:
Unnecessary

Reply:
Bert, the reason I started adding these "make sure the exception was thrown for the expected reason" checks was in case ClientError gets raised for other reasons than the one we expected.  I recently had a unit test like this one happily passing, despite there being a fault in the unit-under-test that resulted in ClientErrors being raised for a different reason than the one expected.

Let me know if you object to me leaving this here.

Reply:
My bad, I should have been more explicit. The assert should definitely stay - the `pass` statement isn't necessary :)